ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java

```diff
@@ -100,7 +100,9 @@ private void putBlob(MessageInfo messageInfo, ByteBuffer messageBuf, long size)
     // May need to encrypt buffer before upload
     if (performUpload) {
       BlobId blobId = (BlobId) messageInfo.getStoreKey();
-      cloudDestination.uploadBlob(blobId, size, new ByteBufferInputStream(messageBuf));
+      CloudBlobMetadata blobMetadata = new CloudBlobMetadata(blobId, messageInfo.getOperationTimeMs(), messageInfo.getSize());
+      blobMetadata.setExpirationTime(messageInfo.getExpirationTimeInMs());
+      cloudDestination.uploadBlob(blobId, size, blobMetadata, new ByteBufferInputStream(messageBuf));
```

Comment:
Add metrics or log for upload not success cases. To help us be aware of potential race conditions. Same to deleteBlob and updateBlobExpiration.

Reply:
Absolutely.  Am planning to add metrics to both Store and Destination classes in next PR.

Reply:
Filed AMBRY-6093 for this, to be done in either this sprint or next.